service-front/module/Application/src/Module.php

```diff
@@ -156,6 +156,27 @@ public function getServiceConfig()
                     return new Container('UserDetails');
                 },
 
+                // Creates new container to store additional session information
+                'PersistentSessionDetails' => function (ServiceLocatorInterface $sm) {
+                    // Ideally this should all be be contained within a helper class to manage the var's on the container for more type safety
+                    $sessionDetails = new Container('SessionDetails');
+
+                    // breadcrumb so we can determine user's last visited route.
+                    $sessionDetails->currentRoute = $sm->get('Application')->getMvcEvent()->getRouteMatch()->getMatchedRouteName();
+
+                    if (!isset($sessionDetails->previousRoute)) {
+                        $sessionDetails->previousRoute = $sessionDetails->currentRoute;
+                    }
+
+                    if ($sessionDetails->routeStore !== $sessionDetails->previousRoute) {
```

Comment:
No, it's purely used as a switch. So we're not just overwriting the previous with the current route. I thought it better to do it this way, with a comparison. Rather than use a bool to do the switching.